util/log/log.go

```diff
@@ -34,74 +43,173 @@ func FatalOnPanic() {
 	}
 }
 
+func logKV(buf *bytes.Buffer, kvs ...[]interface{}) {
+	var i int
+	var s string
+	var kv []interface{}
+	for kvi := range kvs {
+		kv = kvs[kvi]
+		l := len(kv)
+		for i = 0; i < l; i++ {
+			if i%2 == 0 {
+				if kvi != 0 || i != 0 {
+					buf.WriteString(" ")
+				}
+			} else {
+				buf.WriteString("=")
+			}
+			switch kt := kv[i].(type) {
+			case string:
+				s = kt
+			case fmt.Stringer:
+				s = kt.String()
+			default:
+				s = fmt.Sprintf("%v", kt)
+			}
+			if i%2 == 0 {
+				buf.WriteString(s)
+			} else {
+				buf.WriteString(strconv.Quote(s))
+			}
+		}
+		if i%2 != 0 {
+			panic(fmt.Sprintf("odd number of key-value pairs passed: %v", kv))
+		}
+	}
+	buf.WriteString("\n")
+}
+
+func kvPrint(kvs ...[]interface{}) func(*bytes.Buffer) {
+	return func(buf *bytes.Buffer) {
+		logKV(buf, kvs...)
+	}
+}
+
+func headerKV(sev clog.Severity, depth int, msg string) []interface{} {
+	ret := make([]interface{}, 10)
+	ret[0], ret[1] = "level", clog.SeverityName[sev]
+	ret[2], ret[3] = "time", time.Now().Format(time.RFC3339)
+	ret[4], ret[6] = "file", "line"
+	ret[5], ret[7] = clog.Caller(depth + 1)
+	ret[8], ret[9] = "msg", msg
+	return ret
+}
+
+func logDepth(ctx context.Context, depth int, sev clog.Severity, msg string, kvs []interface{}) {
+	// TODO(tschottdorf): logging hooks should have their entry point here.
+	clog.PrintWith(sev, depth+1,
+		kvPrint(headerKV(sev, depth+1, msg), contextKV(ctx), kvs))
+}
+
+// Infoc logs to the WARNING and INFO logs. It extracts values from the
+// Field keys specified in this package and logs them along with the given
+// message and any additional pairs specified as consecutive elements in kvs.
+func Infoc(ctx context.Context, msg string, kvs ...interface{}) {
+	logDepth(ctx, 1, clog.InfoLog, msg, kvs)
+}
+
 // Info logs to the INFO log.
-// Arguments are handled in the manner of fmt.Print; a newline is appended if missing.
-var Info = clog.Info
+// Arguments are handled in the manner of fmt.Print; a newline is appended.
+func Info(args ...interface{}) {
+	logDepth(nil, 1, clog.InfoLog, fmt.Sprint(args...), nil)
```

Comment:
It's not recommended since contexts are usually passed around a lot, but here it's consciously saying "there is no context", so I think it's ok.